### ISSUE TITLE
add a value to "Default" expiry

### ISSUE THREAD

**Issue by tchbla** [Luzifer/ots#153]
I think we shold provide the default expiry value as text next to the default button (or preselect the value that is the default one).
As I understand from the code, the default value is equal to the configured max value.  
I just added a frontend hack that adds the text from the last element of the select list behind the default label.
![image](https://github.com/Luzifer/ots/assets/148860078/50bf362e-99e4-4940-a4d5-e4d5d3002391)  

Maybe an option for defining the default value within specification is also a nice feature.  
For example we want to configure the shortest availible as a default and allow to manual set longer expiry periods.

Since my hack is just some dirty JS there is no PR attached to this request - if I find some spare time I consider to add one 😄 .

**Comment by tchbla** [comment]
In addition I suggest trimming the newline and spaces before adding the values into the html dom.
So that these spaces and newlines become removed 😉 
![image](https://github.com/Luzifer/ots/assets/148860078/3222a9d4-115f-4856-84a6-82bee57b4c3e)

**Comment by Luzifer** [comment]
> In addition I suggest trimming the newline and spaces before adding the values into the html dom.
> So that these spaces and newlines become removed 😉

Autoformatter says: No.

**Comment by Luzifer** [comment]
> Additionally, it might be beneficial to include an option in the specification for defining the default value.

The default value is set through the `SECRET_EXPIRY` environment variable and is used by the API when there is no value for expiry set which is exactly what the frontend choice "Default Expiry" does: It does not specify any value and therefore leaves the choice to the server.

Having two "defaults" is kinda confusing. On the one hand configuring the default expiry and then also configuring the default-frontend-expiry which is not the value of "Default Expiry" and also **must** match one of the software-defined choices not deriving by one second.

> I suggest displaying the default expiry value as text next to the default button, or alternatively, preselecting the default value.

That's not that easy: How would you display `SECRET_EXPIRY=263543`? In Go notation that's 73h12m23s or assuming 86400s as a day "3 Days, 1 Hour, 12 Minutes, 23 Seconds". Sure, that's a constructed case but something to be taken into account. The full field text would then be "Default Expiry (3 Days, 1 Hour, 12 Minutes, 23 Seconds)" which is "a little" too long…

In this case also the default value cannot be selected as it simply does not exist in the dropdown.

> Based on my code analysis, the default value appears to be identical to the configured maximum value.

The default value is `null`, which leaves the choice to the server (which can be infinite storage of the secret). Which also is difficult to display ("Default Expiry (Infinity)"? - too confusing for users)…

**Comment by mshedsilegx** [comment]
@Luzifer What would be the best approach to just display the value of default expiry or the related custom value if overwritten, in the frontend UI ?

**Comment by Luzifer** [comment]
Well basically part of https://github.com/Luzifer/ots/pull/161 - Just without the unrelated changes (see my last comment before closing the PR). Will have a look at this myself, just need to find the time.

**Comment by mshedsilegx** [comment]
Yes, that's exactly what I was looking for. Appreciate the response.

**Comment by fblz** [comment]
The default expiry setting on the web interface was confusing for 12 out of 15 people I had test OTS.
They all asked what it was and were surprised that it represents the maximum value.

I would separate the maximum and default values for expiry.
The default for the web interface could either be a separate option or the first item in the list of available options.
The maximum value would become its own option.
This way I can set a maximum of 72h, a default of 12h and show any value between 1 second and maximum to the users.
The new default can also be used for other consumers of the API.

A high maximum value is desirable for some use cases, like colleague is on holiday till Monday and you want to send them password on Friday before going on holiday yourself. I still would want the default to be much lower than.
